### DISCUSSION TITLE
What is the preferred way to convert from Vec<T> to Arrays where T is a complex struct?

### DISCUSSION THREAD

**Discussion by scimas** [jorgecarleitao/arrow2#1265] [Q&A] [answered]
As an example, let's say T is
```rust
struct Person {
    name: String,
    age: u8,
    apartment: Option<u32>,
}
```
How would I go about converting a `Vec<Person>` to a chunk of `Utf8Array, PrimitiveArray<u8>, PrimitiveArray<u32>` (I assume those are the correct arrow types for the respective fields)?

For a few fields it is not too verbose to create the `Array`s manually, but for larger structs I'm not sure how to achieve this.

**Comment by jorgecarleitao** [comment, marked as answer]
Using https://github.com/DataEngineeringLabs/arrow2-convert awesome library by @ncpenke :)

**Reply by scimas** [reply]
That's awesome indeed!
